contracts/0.8.25/vaults/VaultHub.sol

```diff
@@ -433,55 +434,67 @@ contract VaultHub is PausableUntilWithRoles {
+                    // uint256 mintedStETH = (totalMintedShares * _postInternalEther) / _postInternalShares;
+                    (totalMintedShares * _postInternalEther * TOTAL_BASIS_POINTS)
+                        / (_postInternalShares *  (TOTAL_BASIS_POINTS - socket.reserveRatioBP)),
                     CONNECT_DEPOSIT
                 );
             }
         }
     }
 
     /// @dev impossible to invoke this method under negative rebase
-    function _calculateTreasuryFees(
-        VaultSocket memory _socket,
-        uint256 _postTotalSharesNoFees,
-        uint256 _postTotalPooledEther,
-        uint256 _preTotalShares,
-        uint256 _preTotalPooledEther
-    ) internal view returns (uint256 treasuryFeeShares) {
-        IStakingVault vault_ = IStakingVault(_socket.vault);
+    function calculateVaultTreasuryFees(
+        uint256 _reportValuation,
+        VaultSocket memory socket,
 
-        uint256 chargeableValue = Math256.min(
-            vault_.valuation(),
-            (_socket.shareLimit * _preTotalPooledEther) / _preTotalShares
+        uint256 _preTotalShares,
+        uint256 _preTotalPooledEther,
+        uint256 _postInternalShares,
+        uint256 _postInternalEther,
+        uint256 _sharesToMintAsLidoCoreFees
+    ) public pure returns (uint256 treasuryFeeShares) {
+        uint256 mintableRatio = (TOTAL_BASIS_POINTS - socket.reserveRatioBP);
+        // we are charging fees over the mintable part of the vault's valuation
+
+        uint256 chargeableValuation = Math256.min(
+            // internal share rate is equal to total share rate, so we can use internal values here
+            socket.shareLimit * _postInternalEther / _postInternalShares,
+            _reportValuation * mintableRatio / TOTAL_BASIS_POINTS
         );
 
-        // treasury fee is calculated as a share of potential rewards that
-        // Lido curated validators could earn if vault's ETH was staked in Lido
-        // itself and minted as stETH shares
-        //
-        // treasuryFeeShares = value * lidoGrossAPR * treasuryFeeRate / preShareRate
-        // lidoGrossAPR = postShareRateWithoutFees / preShareRate - 1
-        // = value  * (postShareRateWithoutFees / preShareRate - 1) * treasuryFeeRate / preShareRate
+        // We are charging potential rewards that is chargeableValuation * LidoCoreGrossRewardRate
+        // LidoCoreGrossRewardRate is the Lido core protocol validation reward rate for the day without fees charged
+        // LidoCoreGrossRewardRate = shareRateAfterReportWithoutFeesCharged / shareRateBeforeReport - 1
+        // shareRateAfterReportWithoutFeesCharged = _postInternalEther / (_postInternalShares - _sharesToMintAsLidoCoreFees)
+        // shareRateBeforeReport = _preTotalPooledEther / _preTotalShares
+        uint256 chargeableValuationWithPotentialRewards = chargeableValuation * _postInternalEther * _preTotalShares
```

Comment:
let's improve the explaining part here and clearly indicate how it differs from the Lido Core protocol fee